src/vue/forms/UploadPreIssuanceForm.vue

```diff
@@ -0,0 +1,190 @@
+
+import { vuexTypes } from '@/vuex'
+import { mapGetters } from 'vuex'
+
+export default {
+  name: 'create-issuance-form',
+  components: {
+    FileField
+  },
+  mixins: [FormMixin],
+  props: {
+    isShown: { type: Boolean, default: true }
+  },
+  data: _ => ({
+    documents: {
+      preIssuance: null
+    },
+    userOwnedTokens: null,
+    issuances: []
+  }),
+  computed: {
+    ...mapGetters([
+      vuexTypes.account
+    ])
+  },
+  watch: {
+    'documents.preIssuance': async function (value) {
+      if (value) {
+        const extracted = await FileHelper.readFileAsText(value.file)
+        try {
+          this.parsePreIssuances(JSON.parse(extracted).issuances)
+        } catch (e) {
+          console.error(e)
+          Bus.error('errors.file-corrupted')
+        }
+      }
+    }
+  },
+  async created () {
+    this.disableForm()
+    await this.loadUserOwnedTokens()
+  },
+  methods: {
+    async loadUserOwnedTokens () {
+      const response = await Sdk.horizon.account.getDetails(
+        this[vuexTypes.account].id
+      )
+      this.userOwnedTokens = response.data.map(balance =>
+        balance.assetDetails).filter(asset =>
+        asset.owner === this[vuexTypes.account].accountId)
+    },
+    async submit () {
+      this.disableForm()
+      try {
+        const issuance = this.issuances[0]
+        if (this.isNullAssetSigner(issuance.asset)) {
+          Bus.error('errors.null-asset-signer')
+          this.enableForm()
+          return
+        }
+        const operation = base.PreIssuanceRequestOpBuilder
+          .createPreIssuanceRequestOp({
+            request: issuance.xdr
+          })
+        await Sdk.horizon.transactions.submitOperations(operation)
+        Bus.success('status-message.pre-issuance-uploaded')
+        this.closeForm()
+      } catch (e) {
+        console.error(e)
+        ErrorHandler.process(e)
+      }
+      this.enableForm()
+    },
+    closeForm () {
+      this.$emit('update:isShown', false)
+    },
+    parsePreIssuances (issuances) {
+      const items = issuances.map(item => {
+        const _xdr = base.xdr.PreIssuanceRequest
+          .fromXDR(item.preEmission, 'hex')
+        const result = base.PreIssuanceRequest.dataFromXdr(_xdr)
+        result.xdr = _xdr
+        result.isUsed = item.used
+        return result
+      }).filter(item => {
+        return !this.issuances.find(el => el.reference === item.reference)
+      })
+      for (let i = 0; i < items.length; i++) {
+        const assetCode = items[i].asset
+        if (!this.isAssetDefined(assetCode)) {
+          Bus.error('errors.pre-issuance-not-allowed')
+          this.issuances = []
+          this.disableForm()
+          return
+        }
+      }
+      this.issuances = items
+      this.enableForm()
+    },
+    isAssetDefined (assetCode) {
+      return Boolean(this.userOwnedTokens
+        .filter(item => item.code === assetCode).length
+      )
+    },
+    isNullAssetSigner (asset) {
```

Comment:
I'm not sure we need this check. If asset have no signer, core will invalidate every transaction with pre-issuances on such asset.

Reply:
Resolved, removed